src/controllers/userProfileController.js

```diff
@@ -276,9 +276,12 @@ const userProfileController = function (UserProfile) {
   const putUserProfile = async function (req, res) {
     const userid = req.params.userId;
     const isRequestorAuthorized = !!(
-      canRequestorUpdateUser(req.body.requestor.requestorId, userid)
-      && ((await hasPermission(req.body.requestor, 'putUserProfile'))
-        || req.body.requestor.requestorId === userid)
+
+      canRequestorUpdateUser(req.body.requestor.requestorId, userid) && (
+        await hasPermission(req.body.requestor, 'putUserProfile')
+        || req.body.requestor.requestorId === userid || await hasPermission(req.body.requestor, 'postTimeEntry')
+      )
```

Comment:
`postTimeEntry` should not allow editing entire user profiles.  Separate this. It would be best if time entries had their own API.